app/scripts/modules/core/src/presentation/forms/fields/FormikFormField.tsx

```diff
@@ -70,13 +82,56 @@ function FormikFormFieldImpl<T = any>(props: IFormikFormFieldImplProps<T>) {
     messageNode,
   };
 
+  const [inputMode, setInputMode] = React.useState(SpelAwareInputMode.DEFAULT);
+
+  const modeToInput: { [k in SpelAwareInputMode]: React.ComponentType<IFormInputProps> } = {
+    [SpelAwareInputMode.DEFAULT]: input,
+    [SpelAwareInputMode.FREEFORM]: SimpleSpelInput,
+  };
+
+  const freeformInputEnabled = firstDefined(
+    get(props, 'spelConfig.freeformInputEnabled', null),
+    get(useContext(FormikSpelContext), 'freeformInputEnabled', null),
```

Comment:
`useContext(FormikSpelContext)` should never be nullish, so how about:
```suggestion
    useContext(FormikSpelContext).freeformInputEnabled,
```
I would also suggest extracting the useContext to a top level function variable.   my reasoning (opinion alert) is that it's easier to parse what hooks are in use if they're all at the top level.